src/game/client/draw/w3dtreedraw.cpp

```diff
@@ -0,0 +1,117 @@
+        { "MoveOutwardTime", &INI::Parse_Duration_Unsigned_Int, 0, offsetof(W3DTreeDrawModuleData, m_moveOutwardTime) },
+        { "MoveInwardTime", &INI::Parse_Duration_Unsigned_Int, 0, offsetof(W3DTreeDrawModuleData, m_moveInwardTime) },
+        { "MoveOutwardDistanceFactor", &INI::Parse_Real, 0, offsetof(W3DTreeDrawModuleData, m_moveOutwardDistanceFactor) },
+        { "DarkeningFactor", &INI::Parse_Real, 0, offsetof(W3DTreeDrawModuleData, m_darkeningFactor) },
+        { "ToppleFX", &INI::Parse_FX_List, 0, offsetof(W3DTreeDrawModuleData, m_toppleFX) },
+        { "BounceFX", &INI::Parse_FX_List, 0, offsetof(W3DTreeDrawModuleData, m_bounceFX) },
+        { "StumpName", &INI::Parse_AsciiString, 0, offsetof(W3DTreeDrawModuleData, m_stumpName) },
+        { "KillWhenFinishedToppling", &INI::Parse_Bool, 0, offsetof(W3DTreeDrawModuleData, m_killWhenFinishedToppling) },
+        { "DoTopple", &INI::Parse_Bool, 0, offsetof(W3DTreeDrawModuleData, m_doTopple) },
+        { "InitialVelocityPercent",
+            &INI::Parse_Percent_To_Real,
+            0,
+            offsetof(W3DTreeDrawModuleData, m_initialVelocityPercent) },
+        { "InitialAccelPercent", &INI::Parse_Percent_To_Real, 0, offsetof(W3DTreeDrawModuleData, m_initialAccelPercent) },
+        { "BounceVelocityPercent",
+            &INI::Parse_Percent_To_Real,
+            0,
+            offsetof(W3DTreeDrawModuleData, m_bounceVelocityPercent) },
+        { "MinimumToppleSpeed",
+            &INI::Parse_Positive_Non_Zero_Real,
+            0,
+            offsetof(W3DTreeDrawModuleData, m_minimumToppleSpeed) },
+        { "SinkDistance", &INI::Parse_Positive_Non_Zero_Real, 0, offsetof(W3DTreeDrawModuleData, m_sinkDistance) },
+        { "SinkTime", &INI::Parse_Duration_Unsigned_Int, 0, offsetof(W3DTreeDrawModuleData, m_sinkTime) },
+        { "DoShadow", &INI::Parse_Bool, 0, offsetof(W3DTreeDrawModuleData, m_doShadow) }
+    };
+
+    ModuleData::Build_Field_Parse(p);
+    p.Add(dataFieldParse, 0);
+}
+
+W3DTreeDraw::W3DTreeDraw(Thing *thing, ModuleData const *module_data) : DrawModule(thing, module_data), m_transformSet(false)
+{
+}
+
+ModuleData *W3DTreeDraw::Friend_New_Module_Data(INI *ini)
+{
+    W3DTreeDrawModuleData *data = new W3DTreeDrawModuleData();
+
+    if (ini) {
+        ini->Init_From_INI_Multi_Proc(data, W3DTreeDrawModuleData::Build_Field_Parse);
+    }
+
+    return data;
+}
+
+Module *W3DTreeDraw::Friend_New_Module_Instance(Thing *thing, ModuleData const *module_data)
+{
+    return NEW_POOL_OBJ(W3DTreeDraw, thing, module_data);
+}
+
+NameKeyType W3DTreeDraw::Get_Module_Name_Key() const
+{
+    static NameKeyType nk = g_theNameKeyGenerator->Name_To_Key("W3DTreeDraw");
+    return nk;
+}
+
+void W3DTreeDraw::React_To_Transform_Change(const Matrix3D *matrix, const Coord3D *pos, float angle)
+{
+    Drawable *drawable = Get_Drawable();
+
+    if (!m_transformSet && (drawable->Get_Position()->x != 0.0f || drawable->Get_Position()->y != 0.0f)) {

```

Comment:
it is a pity it only checks x and y as otherwise you could have used a Coords3D zero to compare against.